components/brave_wallet/browser/zcash/zcash_orchard_storage.h

```diff
@@ -16,29 +16,62 @@
 #include "base/types/expected.h"
 #include "brave/components/brave_wallet/common/brave_wallet.mojom.h"
 #include "brave/components/brave_wallet/common/zcash_utils.h"
+#include "brave/components/services/brave_wallet/public/mojom/zcash_decoder.mojom.h"
 #include "sql/database.h"
-
-namespace sql {
-class Database;
-}
+#include "sql/statement.h"
 
 namespace brave_wallet {
 
+template <size_t const T>
+base::expected<std::optional<std::array<uint8_t, T>>, std::string>
+ReadSizedBlob(sql::Statement& statement, size_t position) {
+  auto columns = statement.ColumnCount();
+  CHECK(columns >= 0);
+  if (position >= static_cast<size_t>(columns)) {
+    return base::unexpected("Position mismatch");
+  }
+
+  if (statement.GetColumnType(position) == sql::ColumnType::kNull) {
+    return std::nullopt;
+  }
+
+  if (statement.GetColumnType(position) != sql::ColumnType::kBlob) {
+    return base::unexpected("Type mismatch");
+  }
+
+  auto blob = statement.ColumnBlob(position);
+  if (blob.size() != T) {
+    return base::unexpected("Size mismatch");
+  }
+
+  std::array<uint8_t, T> to;
+  base::ranges::copy_n(blob.begin(), to.size(), to.begin());
```

Comment:
Preferred way now is something like
base::SpanWriter(base::span(data)).Write(blob)
or
base::span(to).copy_from(blob)